### PR TITLE
feat: Added support for superuser access for cluster resource

### PR DESCRIPTION
**Pull request by edbamolvyavahare**

### CODE REVIEW

File: pkg/provider/resource_cluster.go [hunk 1]
```diff
@@ -383,6 +384,11 @@ func (c *clusterResource) Schema(ctx context.Context, req resource.SchemaRequest
 				ElementType:         types.StringType,
 				PlanModifiers:       []planmodifier.Set{setplanmodifier.UseStateForUnknown()},
 			},
+
+			"superuser_access": schema.BoolAttribute{
+				MarkdownDescription: "Is superuser access enabled.",
+				Optional:            true,
```
Comment (wai-wong-edb) [inline pkg/provider/resource_cluster.go:390]: This should be a computed field only if we don't set it in the request and expect it in the response

**Review by wai-wong-edb — CHANGES_REQUESTED** [review]

File: pkg/provider/resource_cluster.go (hunk 1 above)
Comment (edbamolvyavahare) [reply pkg/provider/resource_cluster.go:390]: this is a boolean value which user can set to enable/disable

Comment (edbamolvyavahare) [reply pkg/provider/resource_cluster.go:390]: maybe we can update the description to me more specific

Comment (wai-wong-edb) [reply pkg/provider/resource_cluster.go:390]: I see. In that case we should have both optional: true and computed: true. So we can set it and also expect a value back

**Review by wai-wong-edb — APPROVED** [review]
LGTM. Tested with `superuser_access = true` and omitted

File: pkg/provider/resource_cluster.go (hunk 1 above)
Comment (edbamolvyavahare) [reply pkg/provider/resource_cluster.go:390]: done

**Review by wai-wong-edb — APPROVED** [review]

**Pull request merged**